kythe/cxx/indexer/textproto/analyzer.cc

```diff
@@ -927,34 +941,39 @@ absl::Status AnalyzeCompilationUnit(PluginLoadCallback plugin_loader,
         "Unable to find proto message in descriptor pool: ", message_name));
   }
 
-  for (auto& source : file_data_by_path) {
+  // Only recordio format specifies multiline_separator.
+  // Presense of multiline_separator flag indicates it's recordio file format.
+  absl::optional<std::string> multiline_separator =
+      FindArg(&args, "--multiline_separator");
+  bool is_recordio = multiline_separator.has_value();
+  for (auto& [filepath, filecontent] : file_data_by_path) {
     // Use reflection to create an instance of the top-level proto message.
     // note: msg_factory must outlive any protos created from it.
     google::protobuf::DynamicMessageFactory msg_factory;
     std::unique_ptr<Message> proto(msg_factory.GetPrototype(descriptor)->New());
 
     // Parse textproto into @proto, recording input locations to @parse_tree.
     TextFormat::ParseInfoTree parse_tree;
-    {
+    if (!is_recordio) {
```

Comment:
Can this block be moved into the `else` of the conditional below?

Reply:
Done.